### PR TITLE
[Skip CI] Virtual Texturing cache settings

### PR DESCRIPTION
**Pull request by NicoLeyman**
# This PR is dependant on a trunk PR that changes some VT settings API needed by this PR:
https://ono.unity3d.com/unity/unity/pull-request/106456/_/graphics/virtualtexturing/cache-recreation

# Purpose of this PR

> This PR adds a new Virtual Texturing section to the HDRP asset.
From this section the sizes of the caches used by the Streaming Virtual Texturing system can be configured.

# Testing status
QA pass will follow in the next few days.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/VirtualTexturingSettingsUI.cs [hunk 1]
```diff
@@ -0,0 +1,267 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditorInternal;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+using UnityEngine.Rendering.VirtualTexturing;
+
+#if ENABLE_VIRTUALTEXTURES
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/VirtualTexturingSettingsUI.cs:9]: Why is the UI for VT settings guarded here? #if ENABLE_VIRTUALTEXTURES
Just greying out the settings should be sufficient no ?

**Comment by sebastienlagarde** [comment]
Looks ok to me appart from one question, added our QA that have test VT for validation.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/VirtualTexturingSettingsUI.cs (hunk 1 above)
Comment (NicoLeyman) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/VirtualTexturingSettingsUI.cs]: There's no particular reason why we didn't show the settings when VT is disabled.
I've pushed a new commit that changes this to have the settings grayed out. Seems to work correctly but will notify QA about this as well.

File: com.unity.shadergraph/Editor/Drawing/PreviewManager.cs [hunk 2]
```diff
@@ -354,11 +354,11 @@ void AssignPerMaterialPreviewProperties(Material mat, List<PreviewProperty> perM
                             {
                                 // Ensure we always request the mip sized 256x256
                                 int width, height;
-                                UnityEngine.Rendering.VirtualTexturing.System.GetTextureStackSize(mat, stackPropertyId, out width, out height);
+                                UnityEngine.Rendering.VirtualTexturing.Streaming.GetTextureStackSize(mat, stackPropertyId, out width, out height);
```
Comment (marctem) [inline com.unity.shadergraph/Editor/Drawing/PreviewManager.cs:357]: This change is part of another PR and should not be included here. Please remove so this PR is confined to HD only.

Comment (NicoLeyman) [reply com.unity.shadergraph/Editor/Drawing/PreviewManager.cs]: This diffe should now be gone after merging master.

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 3]
```diff
@@ -517,16 +518,16 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed an issue with refraction model and ray traced recursive rendering (case 1198578).
 - Fixed an issue where a dynamic sky changing any frame may not update the ambient probe.
 - Fixed cubemap thumbnail generation at project load time.
-- Fixed cubemap thumbnail generation at project load time. 
+- Fixed cubemap thumbnail generation at project load time.
```
Comment (GraphineCharles) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md:521]: Are these chagnes (and below) nececarry keep the diff as small as possible it seems like a whitespace/newline issue?

Comment (NicoLeyman) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md:521]: From what I can tell these sneaked in when resolving a merge conflict using Github Desktop (which used Atom).
Maybe we want to keep these whitespace changes for consistency ? The file itself does need to be touched by this PR.

Comment (NicoLeyman) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md]: Update: Whitespace changes removed.

**Pull request merged**